proxy/proxy.go

```diff
@@ -470,6 +485,23 @@ func (p *Proxy) makeBackendRequest(ctx *context) (*http.Response, error) {
 	return response, nil
 }
 
+func (p *Proxy) checkBreaker(c *context) (func(bool), bool) {
+	if p.breakers == nil {
+		return nil, true
+	}
+
+	settings, _ := c.stateBag[circuitfilters.RouteSettingsKey].(circuit.BreakerSettings)
+	settings.Host = c.outgoingHost
```

Comment:
we cannot set breakers for <shunt> and <loopback> . I think the "shunt", by its name, already breaks the circuit by itself. For the loopback, not sure, but i believe it is enough to see where it loops to.